src/Helpers/QueryBuilder.php

```diff
@@ -67,6 +62,39 @@ public static function filterModel(object $model, Request $request = null): Lara
             }
         }
 
+        $relatedAllowedFields = [];
+        foreach (explode(',', $request->include ?: '') as $includedItem) {
+            if (! in_array($includedItem = trim($includedItem), $includes)) {
+                continue;
+            }
+
+            $related = $model->$includedItem()->getRelated();
+
+            $relatedAllowedFields = array_merge(
+                $relatedAllowedFields,
+                ModelInfo::forModel($related)->attributes
+                    ->where('hidden', false)
+                    ->where('virtual', false)
+                    ->pluck('name')
+                    ->map(fn ($value) => $includedItem . '.' . $value)
+                    ->flip()
+                    ->toArray()
+            );
+        }
+
+        $allowed = $model::getColumns();
+        $allowedFields = array_flip(
+            array_map(
+                fn ($item) => $model->getTable() . '.' . $item,
+                $allowed->pluck('Field')->toArray()
+            )
+        );
+        $queryBuilder->allowedFields(array_keys(array_merge($allowedFields, $relatedAllowedFields)));
+
+        if (count($includes) > 0) {
```

Comment:
$includes could be null count on null is deprecated

Reply:
array_diff returns an array, this can never be null